middleware/addStudent.js

```diff
@@ -14,39 +14,59 @@ module.exports = async (req, res, next) => {
       contact,
     } = req.body;
 
-    let parentMongo = await Parent.findOne({ postcode: postcode, houseNo });
+    const parentMongo = await Parent.findOne({ postcode: postcode, houseNo });
 
     if (!parentMongo) {
-      parentMongo = new Parent({
+      const parentMongo = new Parent({
         name: parentName,
         postcode,
         houseNo,
         email,
         contact,
       });
-
       await parentMongo.save();
-    }
-
-    const newStudent = new Student({
-      name: studentName,
-      dob,
-      parent: parentMongo,
-      gender,
-    });
+      const newStudent = new Student({
+        name: studentName,
+        dob,
+        parent: parentMongo,
+        gender,
+      });
+      console.log("frist time");
```

Comment:
No console logs in your develop branch.
Console logs are for developing purpose only, so only when working on a feature use it to debug, and after that it has served it purpose